tests/system/pages/BasePage.ts

```diff
@@ -12,28 +13,36 @@ export abstract class BasePage<T extends BasePageElements = BasePageElements> {
   }
 
   async goto(url: string) {
-    await this.page.goto(url);
+    await step(`Navigate to URL: ${url}`, async () => {
+      await this.page.goto(url);
+    });
 
     return this;
   }
 
   public async gotoMain(): Promise<this> {
-    await this.page.goto(this.pageElements.url);
-    await this.page.waitForLoadState('load');
-    await this.page.waitForLoadState('domcontentloaded');
-    await this.page.waitForLoadState('networkidle');
+    await step(`Navigate to main page: ${this.pageElements.url}`, async () => {
```

Comment:
lets make it more generic
```suggestion
    await step(`Navigate to the page: ${this.pageElements.url}`, async () => {
```